### PR TITLE
lib: nrf_modem: add CID socket options

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Add CID socket options.

Signed-off-by: Eivind Jølsgard <eivind.jolsgard@nordicsemi.no>

### CODE REVIEW

File: lib/nrf_modem_lib/nrf91_sockets.c [hunk 1]
```diff
@@ -148,7 +148,7 @@ static int z_to_nrf_optname(int z_in_level, int z_in_optname,
 			*nrf_out_optname = NRF_SO_SEC_HOSTNAME;
 			break;
 		case TLS_CIPHERSUITE_LIST:
-			*nrf_out_optname = NRF_SO_SEC_CIPHERSUITE_LIST;
+			*nrf_out_optname = NRF_SO_TLS_DTLS_CIPHERSUITE_LIST;
```
Comment (lemrey) [inline lib/nrf_modem_lib/nrf91_sockets.c:151]: Should this be `NRF_SO_SEC_CIPHERSUITE_LIST` instead?

Comment (eivindj-nordic) [reply lib/nrf_modem_lib/nrf91_sockets.c:151]: Missed that one, thanks!

**Review by lemrey — APPROVED** [review]

**Review by lemrey — APPROVED** [review]
nrfxlib: [d546872](https://github.com/lemrey/sdk-nrfxlib/commit/d546872b2cacf7244cc10cd0981bfc313f6e1330)

zephyr: [e33cb69](https://github.com/lemrey/sdk-zephyr/commit/e33cb69d8973141c391327d063fe425720377f7c)

**Comment by eivindj-nordic** [comment]
updated manifest with sha's

**Pull request merged**